api/src/main/java/org/openmrs/validator/DrugOrderValidator.java

```diff
@@ -93,58 +97,64 @@ public void validate(Object obj, Errors errors) {
 			if (order.getDrug() == null || order.getDrug().getConcept() == null) {
 				ValidationUtils.rejectIfEmpty(errors, "concept", "error.null");
 			}
```

Comment:
I think you need to change this to accept the case where `getDrug()` is null but `getDrugNonCoded()` is not.